pkg/bitcoin/electrum/electrum_integration_test.go

```diff
@@ -439,27 +438,128 @@ func TestEstimateSatPerVByteFee_Integration(t *testing.T) {
 	}
 }
 
-func newTestConnection(t *testing.T, config Config) bitcoin.Chain {
-	electrum, err := Connect(context.Background(), config)
+func newTestConnection(t *testing.T, config electrum.Config) (bitcoin.Chain, context.CancelFunc) {
+	ctx, cancelCtx := context.WithCancel(context.Background())
+	electrum, err := electrum.Connect(ctx, config)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	return electrum
+	return electrum, cancelCtx
 }
 
-func assertConfirmationsCloseTo(t *testing.T, expected uint, actual uint) {
-	delta := uint(2)
-
+func assertNumberCloseTo(t *testing.T, expected uint, actual uint, delta uint) {
```

Comment:
We could make it a part of `testutils`. Non-blocking, it does not come from this PR.